### PR TITLE
Add designdoc for Safe Rolling Update of StatefulSet

### PR DESCRIPTION
**Pull request by d-kuro**
refs: https://github.com/cybozu-go/moco/issues/622

### CODE REVIEW

**Review by kmdkuk — DISMISSED** [review]
LGTM

**Review by yamatcha — APPROVED** [review]

**Review by yamatcha — APPROVED** [review]

File: docs/designdoc/safe_statefulset_rolling_update.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    partition: 3
+```
+
+In MOCO, `updateStrategy` of the StatefulSet of the created MySQLCluster is fixed to `type: RollingUpdate`.
+This document proposes to add a reconciler that dynamically sets `.partition` to realize safe updating of StatefulSet.
+
+### When creating MySQLCluster
+
+Until MySQLCluster successfully starts for the first time, applies StatefulSet without setting `.partition`.
+This is done to prevent a situation where a user-corrected StatefulSet, due to misconfigurations like image issues, does not roll out due to partition.
```
Comment (masa213f) [inline docs/designdoc/safe_statefulset_rolling_update.md:41]: Please change this description. Now we set the partition when creating.

**Review by masa213f — CHANGES_REQUESTED** [review]

File: docs/designdoc/safe_statefulset_rolling_update.md (hunk 1 above)
Comment (d-kuro) [reply docs/designdoc/safe_statefulset_rolling_update.md]: Thanks, I fixed: https://github.com/cybozu-go/moco/pull/628/commits/111de9712f49d0dec860aa1d0aaaf79f54715257

**Review by masa213f — APPROVED** [review]
LGTM!

**Pull request merged**